src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -88,14 +109,36 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
       isSendingStreams,
       this.context.videoStreamIndex.localStreamDescriptions(),
       // TODO: handle check-in mode, or remove this param
-      true
+      true,
+      compressedSDPOffer
     );
     this.context.logger.info(`sending subscribe: ${JSON.stringify(subscribe)}`);
     this.context.signalingClient.subscribe(subscribe);
 
     const subscribeAckFrame = await this.receiveSubscribeAck();
     this.context.logger.info(`got subscribe ack: ${JSON.stringify(subscribeAckFrame)}`);
+
+    let decompressedText = '';
+    if (subscribeAckFrame.compressedSdpAnswer && subscribeAckFrame.compressedSdpAnswer.length > 0) {
+      decompressedText = this.textCompressor.decompress(
+        subscribeAckFrame.compressedSdpAnswer,
+        this.context.previousSdpAnswerAsString
+      );
+
+      if (decompressedText.length === 0) {
+        this.context.sdpAnswer = '';
+        this.context.previousSdpAnswerAsString = '';
+        this.logAndThrow(`Error occurred while trying to decompress the SDP answer.`);
+      }
+
+      this.context.logger.info(
+        `Decompressed the SDP message from ${subscribeAckFrame.compressedSdpAnswer.length} to ${decompressedText.length} bytes.`
+      );
+      subscribeAckFrame.sdpAnswer = decompressedText;
```

Comment:
Uggh not a fan of mutating input. Can we just check `serverSupportsCompression` to see whether we want compressedSdpAnswer or sdpAnswer?

Reply:
I don't think we are really mutating the application state, but simply replacing the compressed data with the decompressed data.

We refer to the sdpAnswer at many places as we pass across the subscribeFrame object at multiple places. I feel Decompressing the sdpAnswer is the cleanest way, else we would have to make changes at multiple places and pass around the states at multiple places.

Reply:
I think you misunderstood my comment. I just mean not to change `subscribeAckFrame.sdpAnswer` here and not about context.sdpAnswer. I believe we do not use `subscribeAckFrame.sdpAnswer` anywhere else. Basically:
```
if (support server compression) {
         let decompressText = '';
         if (subscribeAckFrame.compressedSdpAnswer?.length > 0) {
             decompressText = this.textCompressor.decompress(
               subscribeAckFrame.compressedSdpAnswer,
               this.context.previousSdpAnswerAsString
          );
         }
         if (decompressedText.length === 0) {
          this.context.sdpAnswer = '';
          this.context.previousSdpAnswerAsString = '';
          this.logAndThrow(`Error occurred while trying to decompress the SDP answer.`);
         }

         this.context.logger.info(
        `Decompressed the SDP message from ${subscribeAckFrame.compressedSdpAnswer.length} to ${decompressedText.length} bytes.`
      );
         this.context.sdpAnswer = decompressText
} else {
      this.context.sdpAnswer = subscribeAckFrame.sdpAnswer;
} 
```